### PR TITLE
Exit installer if directory already exists when trying to clone one of our repos into it

### PR DESCRIPTION
**Pull request by PromoFaux**
**By submitting this pull request, I confirm the following:** 

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Rather than deleting the directory, return 1 so that the installer knows to throw up the error message and halt the install

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -528,8 +528,8 @@ make_repo() {
     printf "  %b %s..." "${INFO}" "${str}"
     # If the directory exists,
     if [[ -d "${directory}" ]]; then
-        # delete everything in it so git can clone into it
-        rm -rf "${directory}"
+        # Return with a 1 to exit the installer. We don't want to overwrite what could already be here in case it is not ours
```
Comment (dschaper) [inline automated install/basic-install.sh:531]: We could `printf` something that says `/var/www/html/admin` exists before returning `1`. Just to have some kind out output to look at when users show us "Contact support" screenshots.

Comment (dschaper) [reply automated install/basic-install.sh:531]: `printf "${directory}"` I mean.

Comment (PromoFaux) [reply automated install/basic-install.sh:531]: Done, see latest commit

**Review by dschaper — APPROVED** [review]

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -528,8 +528,10 @@ make_repo() {
...
+        str="Unable to clone ${remoteRepo} into ${directory}"
```
Comment (DL6ER) [inline automated install/basic-install.sh:532]:
```suggestion
        str="Unable to clone ${remoteRepo} into ${directory}: Directory already existing"
```

Comment (PromoFaux) [reply automated install/basic-install.sh:532]:
```suggestion
        str="Unable to clone ${remoteRepo} into ${directory} : Directory already exists"
```

**Review by DL6ER — APPROVED** [review]

**Review by dschaper — APPROVED** [review]

**Pull request merged**